### PR TITLE
Do not block calls once stopCh is closed

### LINKED ISSUE #47: When the dispatcher is stopped, new calls shouldn't block
[Here](https://github.com/keybase/go-framed-msgpack-rpc/blob/5f85a2a031aa3a8ebd07895a3576fc9f30ea774a/dispatch.go#L166), we will block if the call loop has been stopped.

This will fix it:
```go
select {
    case <-d.stopCh:
        return io.EOF
    case d.callCh <- call:
        return <-call.resultCh
}
```

### PR DESCRIPTION
**Pull request by jzila**
Fixes #47

### CODE REVIEW

**Comment by maxtaco** [comment]
Confirmed that this solved my bug in keybase/client#1577

**Comment by jzila** [comment]
Fred please review :)

**Comment by akalin-keybase** [comment]
Can you add a regression test?

**Comment by jzila** [comment]
Sure

File: dispatch.go [hunk 1]
```diff
@@ -163,8 +163,12 @@ func (d *dispatch) nextSeqid() seqNumber {
 func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res interface{}, u ErrorUnwrapper) error {
 	profiler := d.log.StartProfiler("call %s", name)
 	call := newCall(ctx, name, arg, res, u, profiler)
-	d.callCh <- call
-	return <-call.resultCh
+	select {
+	case d.callCh <- call:
+		return <-call.resultCh
+	case <-d.stopCh:
+		return DisconnectedError{}
```
Comment (akalin-keybase) [inline dispatch.go]: why this one instead of `io.EOF`? Don't care which one, just wondering

**Comment by akalin-keybase** [comment]
lgtm

File: dispatch.go (hunk 1 above)
Comment (jzila) [inline dispatch.go]: Max recommended `DisconnectedError`.

**Comment by maxtaco** [comment]
If the timing is slightly different and you hit a dead connection in the
packetizer then you get that disconnected error.  We can change it to
io.EOF in both cases but it should be be consistent.

**Comment by jzila** [comment]
Thanks Max, I'll file an issue.

**Pull request merged**